modules/creating-oci-infra-resources-services.adoc

```diff
@@ -0,0 +1,31 @@
+* You have logged into your organization’s {oci} account with administrator privileges.
+
+.Procedure
+
+. Create a compartment and ensure you defined your {ocid-first} in the compartment. A compartment is a component where you can organize and isolate your cloud resources. After you create a compartment, Oracle automatically assigns an {ocid} to your organization’s account. An administrator can access all compartments tagged to your organization’s {oci} account. See link:https://docs.oracle.com/en-us/iaas/Content/Identity/Tasks/managingcompartments.htm#ariaid-title5[Creating compartments] in the Oracle documentation. 
+
+. Create a virtual cloud network (VCN). A compute instance, load balancer, and other resources need this network infrastructure to connect to each other over an internet connection. To establish an on-premise network you must manually create subnets, gateways, routing rules, and security policies. See link:https://docs.oracle.com/en-us/iaas/Content/Network/Tasks/create_vcn.htm[Creating a VCN] in the Oracle documentation. 
+
+. Create a network security group (NSG) in your VCN. You can use the NSG to establish advanced security rules for your network. You must locate the NSG in your compartment, so that certain groups can access network resources. See link:https://docs.oracle.com/en-us/iaas/Content/Network/Concepts/create-nsg.htm[Creating an NSG] in the Oracle documentation. 
+
+. Create a dynamic group that hosts compute instances. After you create the dynamic group, you can then create a policy statement that defines rules for your cluster environment. This statement sets the precedent  for each compute instance to join your {product-title} cluster as a self-managed node. See link:https://docs.oracle.com/en-us/iaas/Content/ContEng/Tasks/contengdynamicgrouppolicyforselfmanagednodes.htm[Creating a dynamic group and a policy for self-managed nodes]. 
+
+. Create a policy statement. See link:https://docs.oracle.com/en-us/iaas/Content/Identity/Tasks/managingpolicies.htm[Managing policies] in the Oracle documentation. You must create a policy so that your administrator can grant access to your groups, users, or resources that operate in your network.
+
+. Create a load balancer, so that you can provide automated traffic distribution on your VCN. See link:https://docs.oracle.com/en-us/iaas/Content/Balance/Tasks/managingloadbalancer_topic-Creating_Load_Balancers.htm[Creating a load balancer] in the Oracle documentation. 
+
+. Add a record to a Domain Name System (DNS), so that you Oracle’s edge-network can maintain your domain’s DNS queries. See link:https://docs.oracle.com/en-us/iaas/Content/DNS/Tasks/record-add.htm[Adding a record to a DNS zone] in the Oracle documentation. 
```

Comment:
@pawanpinjarkar . Only one DNS record per cluster?

Reply:
There are actually 3 as mentioned [here](https://github.com/oracle-quickstart/oci-openshift/blob/main/infrastructure.tf#L623-L657)

1.  `api-int.oci-cluster.demo.oci-rhelcert.edge-sro.rhecoeng.com`  Targets the private IP of the load balancer
2. `api.oci-cluster.demo.oci-rhelcert.edge-sro.rhecoeng.com` Targets the public IP of the load balancer
3. `*.apps.oci-cluster.demo.demo.oci-rhelcert.edge-sro.rhecoeng.com` Targets  the public IP of the load balancer

Also, same is documented in [this](https://docs.google.com/document/d/1y31vGcLVDp4c7Z1-LLefR1JqCtat9mTHBu7bUca5L3A/edit) google doc

Reply:
Updated but some further clarification is needed as to why a customer needs to add these records. Also will the records exist on OCI? No customer intervention to create the records.

Reply:
These records won't be present automatically unless using the terraform script which Oracle is still developing for the agent installer. These records need to be created before the instances boot to perform cluster installation smoothly. OpenShift 4.x requires the DNS records `api.$cluster_name.$base_domain` and `api-int.$cluster_name.$base_domain` to point to the Load balancer targeting the control plane machines. Both records must be resolvable from all the nodes within the cluster.  More info about [creating API and Ingress API records](https://github.com/openshift/installer/tree/master/docs/user/openstack#create-api-and-ingress-dns-records).

Reply:
`api.<cluster name>.<base domain>` points to APIVIP
`api-int.<cluster name>.<base domain>` points to APIVIP
`*.apps.<cluster name>.<base domain>` points to IngressVIP

Reply:
Updated.